libraries/app/api.cpp

```diff
@@ -302,26 +302,31 @@ namespace graphene { namespace app {
        const auto& db = *_app.chain_database();
        FC_ASSERT( limit <= 100 );
        vector<operation_history_object> result;
-       const auto& stats = account(db).statistics(db);
-       if( stats.most_recent_op == account_transaction_history_id_type() ) return result;
-       const account_transaction_history_object* node = &stats.most_recent_op(db);
+       account_statistics_object stats;
+       try { stats = account(db).statistics(db); } catch(...) { return result; }
+       const account_transaction_history_object* node;
+       try { node = &stats.most_recent_op(db); } catch(...) { return result; }
        if( start == operation_history_id_type() )
           start = node->operation_id;
+       if( start.instance.value > node->operation_id.instance.value )
+          start = node->operation_id;
 
-       while(node && node->operation_id.instance.value > stop.instance.value && result.size() < limit)
+       const auto& hist_idx = db.get_index_type<account_transaction_history_index>();
+       const auto& by_op_idx = hist_idx.indices().get<by_op>();
+       auto index_start = by_op_idx.begin();
+       auto itr = by_op_idx.lower_bound(boost::make_tuple(account, start));
+
+       while(itr != index_start && itr->operation_id.instance.value > stop.instance.value && result.size() < limit)
```

Comment:
I think it should be itr->operation_id.instance.value >= stop.instance.value (stop is inclusive).

Reply:
stop is not inclusive unless stop is `1.11.0`. the change will break compatibility: https://github.com/bitshares/bitshares-core/blob/master/tests/tests/history_api_tests.cpp#L71

Reply:
It doesn't make sense to have a parameter that is sometimes inclusive and sometimes it isn't. It also means you can either retrieve both 1.11.1 and 1.11.0, or neither, but not 1.11.1 alone.

API documentation says this:
> @param stop ID of the earliest operation to retrieve

Which IMO clearly states that it is meant to be inclusive. IMO this should be fixed. Clients have to handle a list of results anyway and shouldn't break.

Reply:
@pmconrad because the design and/or first implementation of this API is flawed (ambiguous `stop` parameter). Current situation is, most clients in most circumstances (I guess) consider `stop` as not inclusive. We don't want to break them or ask all of them to change. However, if `stop` is always not inclusive, we're unable to fetch `1.11.0` object. So we made a compromise here: if `stop` is `1.11.0` and if the account has `1.11.0` and if result size didn't reach `limit`, we append `1.11.0` to the result (I admit it's ugly to an extent). If the documentation doesn't imply this, we should correct the documentation.

Reply:
I would guess that most clients always use stop==0 and don't even think about if it's inclusive or not. :-)
I think it's extremely ugly, therefore I'd straighten it up.

But I don't particularly care, do what you think is best.

Reply:
A wallet software that shows histories to the user may be always using `0` as `stop`. However a client script for example that detects new deposits is perhaps using a dynamic value as `stop`, which is more likely to be broken if we changed the behavior.

Reply:
i think we should remain compatible if possible in this case. the call is so common, used by probably every bitshares client around in different ways that i am afraid of business loss by a change in the output.